pkg/api/persistentvolume/util.go

```diff
@@ -33,12 +33,5 @@ func DropDisabledFields(pvSpec *api.PersistentVolumeSpec, oldPVSpec *api.Persist
 }
 
 func hasExpansionSecrets(oldPVSpec *api.PersistentVolumeSpec) bool {
-	if oldPVSpec == nil || oldPVSpec.CSI == nil {
-		return false
-	}
-
-	if oldPVSpec.CSI.ControllerExpandSecretRef != nil {
-		return true
-	}
-	return false
+	return oldPVSpec != nil && oldPVSpec.CSI != nil && oldPVSpec.CSI.ControllerExpandSecretRef != nil
```

Comment:
these aren't actually equivalent though?

Reply:
Aren't they logically equal? I don't see difference. 
It returns true when ControllerExpandSecretRef is not nil and false for any other case.

Reply:
IMO it's much more clear and easier to maintain as it was, but like I said before, that's just my opinion.  Personally I prefer more lines of code for the sake of clarity and maintainability but that's just me.

Reply:
I see. Actually it is bunch of null check, if golang had something like in c# for ex. oldPVSpec?.CSI?.ControllerExpandSecretRef != nil the code would be like that. 

In general I prefer to use guard clauses, but it is not neccessary to use it like that in this code, I think.